semantic-core/src/Data/Core/Parser.hs

```diff
@@ -48,65 +50,79 @@ core :: (TokenParsing m, Monad m) => m (Term Core User)
 core = expr
 
 expr :: (TokenParsing m, Monad m) => m (Term Core User)
-expr = atom `chainl1` go where
-  go = choice [ (Core....) <$ dot
-              , (Core.$$)  <$ notFollowedBy dot
-              ]
+expr = ifthenelse <|> lambda <|> rec <|> load <|> assign
```

Comment:
I’ve altered the grammar for two reasons:

1. To accommodate the changes to `Core`, e.g. by adding records & removing edges.
2. To correct the precedence and parsing of a bunch of expressions such that roundtripping via the pretty-printer works correctly.

The latter changes, alongside changes to the pretty-printer and the precedences given for the infix constructors, mean that we now handle all of our syntax consistently whether constructing manually, parsing, or pretty-printing.